rskj-core/src/main/java/co/rsk/core/bc/MiningMainchainView.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.core.bc;
+
+import org.ethereum.core.Block;
+
+import java.util.List;
+
+public abstract class MiningMainchainView implements MainchainView {
+
+    public abstract void addBestBlock(Block bestBlock);
+
+    public abstract List<Block> get();
+
+    public abstract Block getBestBlock();
```

Comment:
Given this is a brand new interface, I suggest having a `Optional<Block>` return type to signal this might return an empty value. Same below.

Reply:
It is not supposed to return an empty value.

Reply:
I thought I saw a null check, nevermind!